client/plugins/userLibrary.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ItemsApiGetItemsRequest,
+  UserLibraryApiGetLatestMediaRequest
+} from '@jellyfin/client-axios';
+import { Plugin } from '@nuxt/types/app';
+
+type GetItemsParams = Omit<ItemsApiGetItemsRequest, 'userId'>;
+type GetLatestMediaParams = Omit<UserLibraryApiGetLatestMediaRequest, 'userId'>;
+
+type UserLibraryType = {
+  getItem: (id: string) => Promise<void>;
+  getItems: (params: GetItemsParams) => Promise<string[]>;
+  getLatestMedia: (params: GetLatestMediaParams) => Promise<string[]>;
+};
+
+declare module '@nuxt/types' {
+  interface Context {
+    $userLibrary: UserLibraryType;
+  }
+
+  interface NuxtAppOptions {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vuex/types/index' {
+  // eslint-disable-next-line -- Current TypeScript rules flag S as unused, but Nuxt requires identical types
+  interface Store<S> {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+const userLibraryPlugin: Plugin = ({ $api, $auth, store }, inject) => {
+  inject('userLibrary', {
+    /**
+     * Fetches an item based on its ID and stores it
+     *
+     * @param {string} id - Item ID
+     */
+    getItem: async (id: string): Promise<void> => {
+      const call = async (): Promise<void> => {
```

Comment:
Not a fan of this.

The check to see if the item is already in the store should be done manually by whatever is calling this, since we might want to override it in some cases.

The structure of it also looks a bit messy.